framework/src/org/checkerframework/common/value/ValueTransfer.java

```diff
@@ -240,48 +246,50 @@ private AnnotationMirror createBooleanAnnotationMirror(List<Boolean> values) {
             Node rightNode,
             NumericalBinaryOps op,
             TransferInput<CFValue, CFStore> p) {
-        List<? extends Number> lefts = getNumericalValues(leftNode, p);
-        List<? extends Number> rights = getNumericalValues(rightNode, p);
         List<Number> resultValues = new ArrayList<>();
-        for (Number left : lefts) {
-            NumberMath<?> nmLeft = NumberMath.getNumberMath(left);
-            for (Number right : rights) {
-                switch (op) {
-                    case ADDITION:
-                        resultValues.add(nmLeft.plus(right));
-                        break;
-                    case DIVISION:
-                        resultValues.add(nmLeft.divide(right));
-                        break;
-                    case MULPLICATION:
-                        resultValues.add(nmLeft.times(right));
-                        break;
-                    case REMAINDER:
-                        resultValues.add(nmLeft.remainder(right));
-                        break;
-                    case SUBTRACTION:
-                        resultValues.add(nmLeft.minus(right));
-                        break;
-                    case SHIFT_LEFT:
-                        resultValues.add(nmLeft.shiftLeft(right));
-                        break;
-                    case SIGNED_SHIFT_RIGHT:
-                        resultValues.add(nmLeft.signedSiftRight(right));
-                        break;
-                    case UNSIGNED_SHIFT_RIGHT:
-                        resultValues.add(nmLeft.unsignedSiftRight(right));
-                        break;
-                    case BITWISE_AND:
-                        resultValues.add(nmLeft.bitwiseAnd(right));
-                        break;
-                    case BITWISE_OR:
-                        resultValues.add(nmLeft.bitwiseOr(right));
-                        break;
-                    case BITWISE_XOR:
-                        resultValues.add(nmLeft.bitwiseXor(right));
-                        break;
-                    default:
-                        throw new UnsupportedOperationException();
+        if (!isIntRange(leftNode, p) && !isIntRange(rightNode, p)) {
```

Comment:
The Range class seems to have support for arithmetic operations.  Why does this routine do nothing if either argument is an IntRange?  (I have the same question for other operations.)